rotkehlchen/interfaces.py

```diff
@@ -70,20 +72,73 @@ def can_query_history(
         """Checks if it's okay to query historical price"""
 
     @abc.abstractmethod
-    def query_historical_price(
+    def _query_historical_price(
             self,
             from_asset: Asset,
             to_asset: Asset,
             timestamp: Timestamp,
     ) -> Price:
-        """An oracle implements this to return a historical price from_asset to to_asset at time.
+        """Implementation specific historical price query logic.
 
         If no price can be found may raise:
         - PriceQueryUnsupportedAsset
         - NoPriceForGivenTimestamp
         - RemoteError
         """
 
+    @abc.abstractmethod
+    def oracle_source(self) -> HistoricalPriceOracle:
+        """Source enum used as key for storing and retrieving cached prices"""
+
+    @abc.abstractmethod
+    def max_cache_seconds(self) -> int:
+        """Maximum age for cached prices before they must be refetched"""
```

Comment:
for what I see this is the max time difference for which a price is valid. At least is the usage you do so this docstring is misleading. Also maybe a better name?

Reply:
done.